client/src/Features/MicrosoftLearn/MicrosoftLearnFilterCore.ts

```diff
@@ -153,7 +153,7 @@ export function getUpdatedURIFromSelectedFilters(
 }
 
 export const getRegexs = (searchTerm: string): RegExp[] => {
-  searchTerm = searchTerm.replace(/\W/gi, '');
+  searchTerm = searchTerm.replace(/([.?*+^$[\]\\(){}|-])/g, '');
```

Comment:
Why did change? How was the previous one behaving differently than this?

Reply:
A bug was reported recently for this. Basically with `searchTerm.replace(/\W/gi, '')`  we are also removing the spaces between the multiple search terms (if any) and combining them to one. So like `azure cloud` --> `azurecloud`, and this was causing unexpected behavior. Have changed this line to replace only the invalid characters, directly now.